src/CodeGeneratorGo.cs

```diff
@@ -169,22 +169,25 @@ private string StagingDir()
 
         private void PreviewCheck(string folder)
         {
-            var isProfile = Settings.Instance.Host.GetValue<bool>("is-profile").Result;
-            if (isProfile) // if a tag is designed to serve profile building, we could omit the preview check
+            var ignore = Settings.Instance.Host.GetValue<bool>("ignore-preview-chk").Result;

```

Comment:
Minor issue, but it seems weird to me that we'll now invoke the generator with `--preview-chk ---ignore-preview-chk` flags.  I understand why we do this, but I think leaving it as `--is-profile` (or possibly shorten it to `--profile`) doesn't look as strange.

Reply:
Yeah, I agree that this is very weird.

We have added more strict checking for the output-folder path, which is making some tags to be weird - for instance some big composite tags (package-composite-v1 in security). The input file of those tags may change constantly and may not have to a profile tag. 

Maybe we should switch to stable when all the swagger is not preview for this kind of tags?

Reply:
So originally this is what I wanted to do (i.e. only generate the package as stable if all swaggers are stable) however this turned out to be problematic (sorry I don't recall the details anymore).  I think there's a higher-level question here about combining preview and stable swaggers in composite packages that needs to be addressed.

Mostly my comment was about naming the switch `--ignore-preview-chk`, as using it along with `--preview-chk` which seems counter-intuitive.

Reply:
Hmm, what if we enabled preview-chk without a switch when building from a tag?

Reply:
Since the `preview-chk` flag is enabled by default in the pipeline (defined in azure-sdk-for-go), I suppose we could enable this as default but introduce another flag to disable this to achieve the same behaviour as previous (ie. optional preview check).
By introducing the native flag of `ignore-preview-chk`, we could hard-code this into the tag descriptions in `readme.go.md` to serve both profile tags and composite tags.

Reply:
Per our offline conversation I think we need more data before changing the semantics of `preview-chk`.

Also, you might want to split the better error messages into a separate PR so you can get those benefits now while we sort the other stuff out.

Reply:
Left me separate the changes of the option flags out of this PR.

This PR will only serve the error message refinement.